spec/app/domain/authentication/authn-azure/authenticator_spec.rb

```diff
@@ -0,0 +1,142 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+
+RSpec.describe 'Authentication::AuthnAzure::Authenticator' do
+
+  include_context "fetch secrets"
+
+  let(:azure_authenticator_name) { "authn-azure-test" }
+
+  let(:azure_authenticator_secrets) do
+    {
+      "provider-uri" => "test-uri"
+    }
+  end
+
+  let (:mocked_verify_and_decode_token) { double("VerifyAndDecodeAzureToken") }
+
+  before(:each) do
+    allow(mocked_verify_and_decode_token).to receive(:call) { |*args|
+      JSON.parse(args[0][:token_jwt]).to_hash
+    }
+  end
+
+  ####################################
+  # request mock
+  ####################################
+
+  let (:authenticate_azure_token_request) do
+    request_body = StringIO.new
+    request_body.puts "jwt={\"xms_mirid\": \"some_xms_mirid_value\", \"oid\": \"some_oid_value\"}"
+    request_body.rewind
```

Comment:
My understanding is that .rewind resets the index to zero? What is the purpose of this here?

Reply:
Honestly I don't remember. It's from authn-oidc.